src/services/activation/utils/index.js

```diff
@@ -0,0 +1,240 @@
+	if (
+		Date.parse(entry.updatedAt) + 1000 * Configuration.get('ACTIVATION_LINK_PERIOD_OF_VALIDITY_SECONDS')
+		< Date.now()
+	) {
+		await deleteEntry(this, entry._id);
+		throw new BadRequest(customErrorMessages.ACTIVATION_LINK_EXPIRED);
+	}
+	if (entry.state !== STATE.NOT_STARTED && entry.state !== STATE.ERROR) {
+		throw new BadRequest(customErrorMessages.ACTIVATION_LINK_INVALID);
+	}
+};
+
+
+const createNewEntry = async (ref, userId, keyword, quarantinedObject) => {
+	try {
+		const entry = await (ref.app || ref).service('activationModel')
+			.create({
+				userId,
+				keyword,
+				quarantinedObject,
+			});
+		return entry;
+	} catch (error) {
+		throw new Error('Could not create a quarantined object.');
+	}
+};
+
+/**
+ * Create new Entry (job) or returns already existing entry if the quarantinedObject is equal
+ * @param {*} ref 					this
+ * @param {ObjectId} userId 		UserId
+ * @param {String} keyword 			keyword
+ * @param {*} quarantinedObject 	quarantinedObject (e.g: email)
+ * @returns {Object}				returns newly created Entry
+ */
+const createEntry = async (ref, userId, keyword, quarantinedObject) => {
+	const entry = await getEntriesByUserId(ref, userId, keyword);
+	if (entry) {
+		if (entry.quarantinedObject === quarantinedObject) {
+			return entry;
+		}
+		await deleteEntry(ref, entry._id);
+	}
+	const newEntry = await createNewEntry(ref, userId, keyword, quarantinedObject);
+	return newEntry;
+};
+
+
+/**
+ * Will send email with informatrion from mail {receiver, subject, content}
+ * also sets mailSent and updatedAt for entry by entryId
+ * @param {*} ref				this
+ * @param {Object} mail			{receiver, subject, content}
+ * @param {ObjectId} entryId	ObjectId of entry
+ */
+const sendMail = async (ref, mail, entry) => {
+	if (!mail || !mail.receiver || !mail.subject || !mail.content || !entry) throw SyntaxError('missing parameters');
+
+	try {
+		await ref.app.service('/mails')
+			.create({
+				email: mail.receiver,
+				subject: mail.subject,
+				content: mail.content,
+			});
+
+		await ref.app.service('activationModel').patch({ _id: entry._id }, {
+			$push: {
+				mailSent: Date.now(),
+			},
+		});
+	} catch (error) {
+		if (entry.mailSent.length === 0) await deleteEntry(ref, entry._id);
+		throw new Error('Can not send mail with activation link');
+	}
+};
+
+/**
+ * Filters a given array of entries. All non validKeys are removed from each object
+ * @param {Array} entries	 array of entries
+ * @param {*} validKeys 	 array of valid Objectkeys
+ * @returns {Array}			 filtered array
+ */
+const filterEntryParamNames = (entries, validKeys) => {
+	if (!Array.isArray(entries)) throw new SyntaxError('entries must be an array');
+	(entries || []).forEach((entry) => {
+		let data = {};
+		Object.keys(entry).forEach((key) => validKeys.includes(key) || delete entry[key]);
+		if (validKeys.includes('quarantinedObject')) {
+			data = entry.quarantinedObject;
+			delete entry.quarantinedObject;
+			entry.data = data;
```

Comment:
The idea behind it was to provide only relevant information for the frontend. Since nothing is done with quarantinedObject in the frontend, but is called data there, I wanted to adapt it accordingly. But yes, this should not happen in this function...